### PR TITLE
[Development][Joseph] initial version

### PR DESCRIPTION
**Pull request by zenthiccc**
please review thank you

### CODE REVIEW

File: client/src/actions/notes.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { FETCH_ALL, CREATE, DELETE } from "../constants/ActionTypes";
+import * as api from "../api";
+
+export const getNotes = () => async (dispatch) => {
+  try {
+    const { data } = await api.fetchNotes();
+    dispatch({ type: FETCH_ALL, payload: data });
+  } catch (error) {
+    console.log(error.message);
```
Comment (sheensantoscapadngan) [inline client/src/actions/notes.js]: console.error(error)

File: client/src/components/Drawer.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+import Drawer from "@material-ui/core/Drawer";
+import Hidden from "@material-ui/core/Hidden";
+import IconButton from "@material-ui/core/IconButton";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import ListItemText from "@material-ui/core/ListItemText";
+import MenuIcon from "@material-ui/icons/Menu";
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
```
Comment (sheensantoscapadngan) [inline client/src/components/Drawer.js]: use { } para ani so 1 line ra dapat

File: client/src/components/Drawer.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  },
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+  },
+  title: {
+    padding: theme.spacing(2),
+  },
+  active: {
+    background: "#808080" //dark mode - #808080, light mode #f4f4f4
```
Comment (sheensantoscapadngan) [inline client/src/components/Drawer.js]: set this color to material ui theme palette

File: client/src/components/Drawer.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            color="inherit"
+            aria-label="open drawer"
+            edge="start"
+            onClick={handleDrawerToggle}
+            className={classes.menuButton}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography noWrap>
+            Today is {format(new Date(), "do MMMM Y")}
```
Comment (sheensantoscapadngan) [inline client/src/components/Drawer.js]: use react-moment for formatting date

Comment (sheensantoscapadngan) [reply client/src/components/Drawer.js]: https://www.npmjs.com/package/react-moment

File: client/src/components/NoteCard.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+const useStyles = makeStyles({
+  avatar: {
+    backgroundColor: (note) => {
+      if (note.category == "Work") {
+        return yellow[700];
+      } else if (note.category == "Money") {
+        return green[700];
+      } else if (note.category == "Todos") {
+        return pink[700];
+      }
```
Comment (sheensantoscapadngan) [inline client/src/components/NoteCard.js]: declare as constants

File: client/src/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./css/index.css";
+import App from "./App";
+import { Provider } from "react-redux";
+import { createStore, applyMiddleware, compose } from "redux";
+import thunk from "redux-thunk";
+import reducers from "./reducers";
+
+const store = createStore(reducers, compose(applyMiddleware(thunk)));
```
Comment (sheensantoscapadngan) [inline client/src/index.js]: separate js

File: client/src/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  <Provider store={store}>
+    <App />
+  </Provider>,
+  document.getElementById("root")
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+// reportWebVitals();
```
Comment (sheensantoscapadngan) [inline client/src/index.js]: remove

File: client/src/reducers/notes.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { FETCH_ALL, CREATE, DELETE } from "../constants/ActionTypes";
+
+export default (notes = [], action) => {
```
Comment (sheensantoscapadngan) [inline client/src/reducers/notes.js]: declare separate initialState variable to hold default values

File: client/src/reducers/notes.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { FETCH_ALL, CREATE, DELETE } from "../constants/ActionTypes";
+
```
Comment (sheensantoscapadngan) [inline client/src/reducers/notes.js:2]: const initialState = []

File: server/index.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+const app = express();
+
+app.use(bodyParser.json());
+app.use(cors());
+
+app.use("/notes", noteRoutes);
+
+const CONNECTION_URL =
+  "mongodb+srv://user123:user123@cluster0.aeekc.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
```
Comment (sheensantoscapadngan) [inline server/index.js]: use config to initialize resources

File: client/public/index.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
```
Comment (sheensantoscapadngan) [inline client/public/index.html]: Erase this and all other unnecessary comments

File: client/public/index.html [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
```
Comment (sheensantoscapadngan) [inline client/public/index.html]: this

**Review by sheensantoscapadngan — CHANGES_REQUESTED** [review]
left comments

File: client/src/actions/notes.js (hunk 1 above)
Comment (zenthiccc) [reply client/src/actions/notes.js]: done

File: client/src/components/Drawer.js (hunk 3 above)
Comment (zenthiccc) [reply client/src/components/Drawer.js]: fixed

File: client/src/components/Drawer.js (hunk 4 above)
Comment (zenthiccc) [reply client/src/components/Drawer.js]: fixed

File: client/src/components/NoteCard.js (hunk 5 above)
Comment (zenthiccc) [reply client/src/components/NoteCard.js]: fixed

File: client/src/index.js (hunk 6 above)
Comment (zenthiccc) [reply client/src/index.js]: done

File: client/src/index.js (hunk 7 above)
Comment (zenthiccc) [reply client/src/index.js]: done

File: client/src/reducers/notes.js (hunk 8 above)
Comment (zenthiccc) [reply client/src/reducers/notes.js]: done

File: client/src/reducers/notes.js (hunk 9 above)
Comment (zenthiccc) [reply client/src/reducers/notes.js:2]: done

File: server/index.js (hunk 10 above)
Comment (zenthiccc) [reply server/index.js]: done

File: client/public/index.html (hunk 11 above)
Comment (zenthiccc) [reply client/public/index.html]: done

File: client/public/index.html (hunk 12 above)
Comment (zenthiccc) [reply client/public/index.html]: done

**Pull request merged**